Hazel/src/Hazel/Renderer/Renderer.h

```diff
@@ -24,7 +24,7 @@ namespace Hazel {
 			glm::mat4 ViewProjectionMatrix;
 		};
 
-		static SceneData* s_SceneData;
+		static Hazel::Scope<SceneData> s_SceneData;
```

Comment:
This doesn't need to be a pointer at all it can just be a normal static member variable, it requires no special ownership

Reply:
I think we should consider keeping it as a smart pointer, simply because Cherno decided to make this a pointer initially. He probably had a reason for doing that. On the other hand he might not have had a specific reason at all. I'm not sure either way.

Reply:
Also based on that logic could the same be said about any changes including your proposed changes?

I'm not certain that is the best logic to follow if the goal is to improve the engine, because functionally there is no difference between the two, just improved performance because your not chasing the pointer.